### DISCUSSION TITLE
Rotation not working properly for Text node

### DISCUSSION THREAD

**Discussion by Arun1947** [sceneview/sceneform-android#332] [Q&A]
Hi,
I need to rotate my text node added in arSceneView like this
90degree, -90 degree, 45 degree, -45 degree

I applied rotation like this
rotationValue=90f ( this is the value applied )
node.localRotation = Quaternion(rotationValue, 90f, -90f, 0f)

I need the rotation only in x axis, Can you help on this?

**Comment by ThomasGorisse** [comment]
Quaternion are not expressed in degrees. You have to use this constructor for Euler Angles:
https://github.com/SceneView/sceneform-android/blob/954162854b2ca84b40c6068b6a314437c17f6506/core/src/main/java/com/google/ar/sceneform/math/Quaternion.java#L42-L52

**Reply by Arun1947** [reply]
Thanks..

**Comment by ThomasGorisse** [comment]
And please use the Discussion section

**Reply by Arun1947** [reply]
sure..!
